src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexMatcher.cs

```diff
@@ -945,24 +936,12 @@ public void InitializeFrom(DfaMatchingState<TSet> dfaMatchingState)
 
                 // If the DFA state is a union of multiple DFA states, loop through all of them
                 // adding an NFA state for each.
-                SymbolicRegexNode<TSet> node = dfaMatchingState.Node.Kind == SymbolicRegexNodeKind.DisableBacktrackingSimulation ?
-                    dfaMatchingState.Node._left! : dfaMatchingState.Node;
-                while (node.Kind is SymbolicRegexNodeKind.OrderedOr)
+                foreach (SymbolicRegexNode<TSet> element in dfaMatchingState.Node.BreakUpAlternation())
```

Comment:
Does "BreakUpAlternation" mean "EnumerateAlternationBranches"?

Reply:
It does essentially and that's a better name.